lib/utils.js

```diff
@@ -114,3 +119,61 @@ export function parseKeystoreId(keyId) {
   }
   return keyId.slice(0, idx);
 }
+
+export async function getEdvConfig({edvClient, profileSigner} = {}) {
+  return edvClient.getConfig({
+    invocationSigner: profileSigner
+  });
+}
+
+export async function getEdvDocument({
+  docId, edvConfig, edvClient, kmsClient, profileSigner
+} = {}) {
+  const {hmac, keyAgreementKey} = edvConfig;
+
+  const doc = new EdvDocument({
+    invocationSigner: profileSigner,
+    id: docId,
+    keyAgreementKey: new KeyAgreementKey({
+      id: keyAgreementKey.id,
+      type: keyAgreementKey.type,
+      invocationSigner: profileSigner,
+      kmsClient
+    }),
+    hmac: new Hmac({
+      id: hmac.id,
+      type: hmac.type,
+      invocationSigner: profileSigner,
+      kmsClient
+    }),
+    client: edvClient
+  });
+  return doc.read();
+}
+
+export async function getProfileSigner({kmsClient, profileAgentRecord} = {}) {
+  const record = JSON.parse(JSON.stringify(profileAgentRecord));
```

Comment:
`klona` use case.  Also, is there a danger of mutation in this function?  I see `profileAgentRecord` is still referenced directly instead of through `record` on L161.

Reply:
Good catch, thank you! I've updated it.